src/shared/Core.Tests/StringExtensionsTests.cs

```diff
@@ -293,5 +293,14 @@ public void StringExtensions_TrimMiddle_String_ComparisonType(string input, stri
             string actual = StringExtensions.TrimMiddle(input, trim, comparisonType);
             Assert.Equal(expected, actual);
         }
+
+        [Theory]
+        [InlineData("FooBar", "foo_bar")]
+        [InlineData("FBBaz", "fb_baz")]
+        [InlineData("Foo", "foo")]
+        public void StringExtensions_ToSnakeCase_Converts_Correctly(string input, string expected)
+        {
+            Assert.Equal(input.ToSnakeCase(), expected);
```

Comment:
The actual and expected arguments are reversed.

```suggestion
            Assert.Equal(expected, input.ToSnakeCase());
```